src/Microsoft.Health.Fhir.Anonymizer.Shared.Core/Utility/DateTimeUtility.cs

```diff
@@ -42,7 +43,14 @@ public static ProcessResult RedactDateNode(ElementNode node, bool enablePartialD
                 if (matchedGroups[s_yearIndex].Captures.Any())
                 {
                     string yearOfDate = matchedGroups[s_yearIndex].Value;
-                    node.Value = IndicateAgeOverThreshold(matchedGroups) ? null : yearOfDate;
+                    if (IndicateAgeOverThreshold(matchedGroups))
+                    {
+                        node.Value = null;
+                    }
+                    else
+                    {
+                        node.Value = PartialDateTime.Parse(yearOfDate);
```

Comment:
It seems there's another type named FhirDateTime. Could we use FhirDateTime here?

Reply:
FhirDateTime will be converted to  PartialDateTime in fhir.

Reply:
Got it. Is PartialDateTime the default type of node.Value for dates, dateTime and instant? Will we change the object type here?

Reply:
Yes, dates, dateTime and instant will be transformed into PartialDateTime.